src/@types/superstruct.d.ts

```diff
@@ -0,0 +1,256 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+// This is a type-shim because superstruct's types do not work with NodeNext module resolution
```

Comment:
This is news to me! Superstruct is usually very good in ESM...

What exactly is the compiler error? Do we need `NodeNext` specifically, or would `ES2022` do the trick?

Reply:
The compiler error is that `superstruct` has no exported properties. TypeScript cannot find the type files.

https://github.com/ianstormtaylor/superstruct/issues/1200

`NodeNext` is the recommended setting for modern node projects. If we set `module` to `ES2022`, then `moduleResolution` will default to `Node10`, which is designed for CommonJS-only importing. At least that's what I gleaned from the TypeScript docs.

From my experience, using `ES*` for `module` is for when you expect to run your code outside node, like in a browser.

Reply:
TIL! Guess I haven't ever read that part of the doc closely enough 😅

Def seems like a bug in Superstruct. I'm fine with leaving the shim in place until that's fixed

---

For my own future reference, the Summary section under [`es2015`, `es2020`, `es2022`, `esnext`](https://www.typescriptlang.org/docs/handbook/modules/reference.html#es2015-es2020-es2022-esnext) specifically says "Do not use for Node.js." Guess I should re-think my tsconfigs in all my other projects too, lol